.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,78 @@
+name: Unit Tests
+on: [push]
+jobs:
+  test:
+    # TODO: Remove continue-on-error after migrating from Travis to GitHub Actions
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        # TODO: Add tip when https://github.com/actions/setup-go/issues/21 is resolved
+        go: [1.16.3]
+        target:
+        - linux-amd64-fmt
+        - linux-amd64-grpcproxy
+        - linux-amd64-coverage
+        - linux-amd64-integration-1-cpu
+        - linux-amd64-integration-2-cpu
+        - linux-amd64-integration-4-cpu
+        - linux-amd64-functional
+        - linux-amd64-unit-4-cpu-race
+        - all-build
+        - linux-amd64-fmt-unit-go-tip-2-cpu
+        - linux-386-unit-1-cpu
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go }}
+    - run: date
+    - env:
+        TARGET: ${{ matrix.target }}
+      run: |
+        echo "${TARGET}"
+        case "${TARGET}" in
+          linux-amd64-fmt)
+            GOARCH=amd64 PASSES='fmt bom dep' ./test.sh
+            ;;
+          linux-amd64-integration-1-cpu)
+            GOARCH=amd64 CPU=1 PASSES='integration' RACE='false' ./test.sh
+            ;;
+          linux-amd64-integration-2-cpu)
+            GOARCH=amd64 CPU=2 PASSES='integration' RACE='false' ./test.sh
+            ;;
+          linux-amd64-integration-4-cpu)
+            GOARCH=amd64 CPU=4 PASSES='integration' RACE='false' ./test.sh
+            ;;
+          linux-amd64-functional)
+            GO_BUILD_FLAGS='-v -mod=readonly' ./build && GOARCH=amd64 PASSES='functional' ./test
+            ;;
+          linux-amd64-unit-4-cpu-race)
+            GOARCH=amd64 PASSES='unit' RACE='true' CPU='4' ./test.sh -p=2
+            ;;
+          all-build)
+            GOARCH=amd64 PASSES='build' ./test.sh
+            GOARCH=386 PASSES='build' ./test.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOOS=darwin GOARCH=amd64 ./build.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOOS=windows GOARCH=amd64 ./build.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOARCH=arm ./build.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOARCH=arm64 ./build.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOARCH=ppc64le ./build.sh
+            GO_BUILD_FLAGS='-v -mod=readonly' GOARCH=s390x ./build.sh
+            ;;
+          linux-amd64-grpcproxy)
+            PASSES='build grpcproxy'  CPU='4' COVER='false' RACE='true' ./test.sh 2>&1 | tee test.log
+            ! egrep "(--- FAIL:|DATA RACE|panic: test timed out|appears to have leaked)" -B50 -A10 test.log
+            ;;
+          linux-amd64-coverage)
```

Comment:
Coverage on its own is huge (30min) and on Travis was running as best effort. 
The PR was 'green' for submit even if it didn't finished.

Is there a way to express it in GitHub actions ? e.g. should be it a workflow on its own, that keeps
`continue-on-error: true` ?

Reply:
Yes, I think we could move it to separate job executed as part of the same workflow. 
In this PR we declare only one job:
```
jobs:
  test:
    ...
```
We could create a separate job for coverage.

Reply:
let's do this in a follow up PR.